### PR TITLE
Add `USE_CONTAINER_BACKUP_DATE`

### PR DESCRIPTION
**Pull request by Minituff**
Requested by https://github.com/Minituff/nautical-backup/issues/380

### Container Backup Date vs Nautical Start Date
Use the precise date for formatting the destination folder
When `false`, use the time Nautical started the backup (not when the container was backed up).

> **Default**: false

```properties
USE_CONTAINER_BACKUP_DATE=true
```

#### Understanding `USE_CONTAINER_BACKUP_DATE`.
Let's say you have the following format: `DEST_DATE_FORMAT=%Y-%m-%d_%H-%M-%S`

Since the timing is so precise <small>(using seconds)</small> we may get a result like this:
```python
-- src
|-- 2024-11-24_14-26-43
|-- 2024-11-24_14-26-44
|-- 2024-11-24_14-28-10
```

But what we actually want is a result like this:
```python
-- src
|-- 2024-11-24_14-26-43
#  The folder that was here is now within in `2024-11-24_14-26-43` folder
|-- 2024-11-24_14-28-10
```

By setting `USE_CONTAINER_BACKUP_DATE=false`, then the date used will be the time Nautical actually started, not the time when each container is processed.
Meaning that even if the containers take a few minutes to backup, the folder format will remain the same for each of them.

### CODE REVIEW

File: app/nautical_env.py [hunk 1]
```diff
@@ -37,6 +37,10 @@ def __init__(self) -> None:
         if self.DEST_DATE_PATH_FORMAT not in ["date/container", "container/date"]:
             self.DEST_DATE_PATH_FORMAT = "date/container"  # Set default
 
+        self.USE_CONTAINER_BACKUP_DATE = True
+        if os.environ.get("USE_CONAINER_BACKUP_DATE", "true").lower() == "false":
```
Comment (mrjk) [inline app/nautical_env.py:41]: This is it, there is a Typo here:
```
USE_CONTAINER_BACKUP_DATE
vs
USE_CONAINER_BACKUP_DATE
```

Comment (Minituff) [reply app/nautical_env.py:41]: Omg, great catch. Let me patch that

File: app/backup.py [hunk 2]
```diff
@@ -481,7 +490,13 @@ def _get_dest_dir(self, c: Container, src_dir_name: str) -> Tuple[Path, str]:
 
     def _format_dated_folder(self, base_dest_dir: Path, folder: str) -> Path:
         """Format the destination folder with the date"""
-        time_format = str(time.strftime(self.env.DEST_DATE_FORMAT))
+
+        # Use Nautical start time as the date for formatting
+        time_format = str(self.start_time.strftime(self.env.DEST_DATE_FORMAT))
+        if self.env.USE_CONTAINER_BACKUP_DATE:
+            # Use current time for formatting
+            self.log_this(f"Using current time for date formatting since DEST_DATE_FORMAT=true", "Trace")
```
Comment (mrjk) [inline app/backup.py:498]: I also guess this is an error on the logging message:
```
DEST_DATE_FORMAT=true => USE_CONTAINER_BACKUP_DATE=true
```
There is a simlar error on line: 476

Comment (Minituff) [reply app/backup.py:498]: 100%, this log was mislabeled

**Pull request merged**